### PR TITLE
Support optional sentry integration

### PR DESCRIPTION
**Pull request by ioparaskev**
Add opt in / out(default)  setting for sentry
Support environmental/hardcoded sentry dsn
Validate sentry dsn is setup if sentry is used
Use default empty dict sentry placeholder for backward settings
compatibility

### CODE REVIEW

**Comment by zmousm** [comment]
Thanks for this.

Would you please:
- Add `raven` (?) to `requirements-optional.txt`.
- Mention it in `docs/installation/requirements.md` under Conditional Dependencies.
- Add a short section about the steps to enable this integration in `docs/installation/install.md`, say, after Extra Apps. For example add this setting and/or that env variable.

File: requirements-optional.txt [hunk 1]
```diff
@@ -1 +1,2 @@
 Mako==1.0.3
+raven
```
Comment (zmousm) [inline requirements-optional.txt]: Would it make sense to limit the range of versions for the package, as applicable for Django 1.8?

**Comment by zmousm** [comment]
See one minor inline comment. Apart from that, I'm ready to merge and I will fine-tune some wording in the docs afterwards.

File: requirements-optional.txt (hunk 1 above)
Comment (ioparaskev) [reply requirements-optional.txt]: I'm not sure if we can set a range, maybe a minimum version would be more preferable?

**Pull request merged**